azuredeploy.json

```diff
@@ -1,158 +1,238 @@
     },
     "resources": [
         {
             "apiVersion": "2016-03-01",
-            "name": "[variables('hostingPlanName')]",
+            "name": "[parameters('hostingPlanName')]",
             "type": "Microsoft.Web/serverfarms",
-            "location": "[resourceGroup().location]",
+            "location": "[parameters('webAppLocation')]",
             "tags": {
                 "displayName": "HostingPlan"
             },
             "sku": {
                 "name": "[parameters('skuName')]",
                 "capacity": 1
             },
-            "properties": {
-                "name": "[variables('hostingPlanName')]"
-            }
+            "properties": {}
         },
         {
             "apiVersion": "2016-03-01",
             "name": "[parameters('webSiteName')]",
             "type": "Microsoft.Web/sites",
-            "location": "[resourceGroup().location]",
+            "location": "[parameters('webAppLocation')]",
             "identity": {
                 "type": "SystemAssigned"
             },
             "dependsOn": [
-                "[variables('hostingPlanName')]"
+                "[resourceId('Microsoft.Web/serverfarms', parameters('hostingPlanName'))]"
             ],
             "tags": {
-                "[concat('hidden-related:', resourceId('Microsoft.Web/serverfarms', variables('hostingPlanName')))]": "empty",
+                "[concat('hidden-related:', resourceId('Microsoft.Web/serverfarms', parameters('hostingPlanName')))]": "empty",
                 "displayName": "Website"
             },
             "properties": {
-                "name": "[parameters('webSiteName')]",
-                "serverFarmId": "[resourceId('Microsoft.Web/serverfarms', variables('hostingPlanName'))]"
-            }
+                "serverFarmId": "[resourceId('Microsoft.Web/serverfarms', parameters('hostingPlanName'))]"
+            },
+            "resources": [
+                {
+                    "apiVersion": "2016-03-01",
+                    "name": "appsettings",
+                    "type": "config",
+                    "location": "[resourceGroup().location]",
+                    "properties": {
+                        "tokenResourceUrl": "[reference(concat('Microsoft.TokenVault/vaults/', parameters('tokenVaultName'), '/services/dropbox/tokens/sampleToken'), '2018-08-01-preview').tokenUri]"
+                    },
+                    "dependsOn": [
+                        "[resourceId('Microsoft.Web/Sites', parameters('webSiteName'))]",
+                        "[resourceId('Microsoft.Web/Sites/sourcecontrols', parameters('webSiteName'), 'web')]",
+                        "[concat('Microsoft.TokenVault/vaults/', parameters('tokenVaultName'), '/services/dropbox/tokens/sampleToken')]"
+                    ]
+                },
+                {
+                    "apiVersion":"2018-02-01",
+                    "name":"web",
+                    "type":"sourcecontrols",
```

Comment:
This is cool. This is all that is needed to deploy the app? 
Are there any side effects for when they want to publish their version to the app? Do they need to disconnect this?

Reply:
this is manual integration - simplified version of https://github.com/projectkudu/kudu/wiki/Deploying-from-GitHub which has webhooks for github to notify for changes.

this is jsut web app pull changes